pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -0,0 +1,135 @@
+	"k8s.io/cloud-provider-gcp/providers/gce"
+	controllersmetrics "k8s.io/component-base/metrics/prometheus/controllers"
+)
+
+// Controller manages GKENetworkParamSet status.
+type Controller struct {
+	gkeNetworkParamsInformer networkinformers.GKENetworkParamSetInformer
+	networkClientset         networkclientset.Interface
+	informerStartFn          func(stopCh <-chan struct{})
+	gceCloud                 *gce.Cloud
+}
+
+// NewGKENetworkParamSetController returns a new
+func NewGKENetworkParamSetController(
+	networkClientset networkclientset.Interface,
+	gkeNetworkParamsInformer networkinformers.GKENetworkParamSetInformer,
+	informerStartFn func(stopCh <-chan struct{}),
+	gceCloud *gce.Cloud,
+) *Controller {
+
+	return &Controller{
+		networkClientset:         networkClientset,
+		gkeNetworkParamsInformer: gkeNetworkParamsInformer,
+		informerStartFn:          informerStartFn,
+		gceCloud:                 gceCloud,
+	}
+
+}
+
+// Run starts an asynchronous loop that monitors and updates GKENetworkParamSet in the cluster.
+func (c *Controller) Run(stopCh <-chan struct{}, controllerManagerMetrics *controllersmetrics.ControllerManagerMetrics) {
+	defer utilruntime.HandleCrash()
+
+	klog.Infof("Starting gkenetworkparamset controller")
+	defer klog.Infof("Shutting down gkenetworkparamset controller")
+	controllerManagerMetrics.ControllerStarted("gkenetworkparamset")
+	defer controllerManagerMetrics.ControllerStopped("gkenetworkparamset")
+
+	c.gkeNetworkParamsInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    gkenetworkparamsetutil.CreateAddGKENetworkParamSetHandler(c.onGKENetworkParamSetAdded),
+		UpdateFunc: gkenetworkparamsetutil.CreateUpdateGKENetworkParamSetHandler(c.onGKENetworkParamSetUpdated),
+	})
+
+	c.informerStartFn(stopCh)
+
+	<-stopCh
+	klog.Infof("done gkenetworkparamset controller")
+}
+
+func (c *Controller) onGKENetworkParamSetAdded(params *networkv1alpha1.GKENetworkParamSet) error {
+	return c.updateGKENetworkParamSet(params)
+}
+func (c *Controller) onGKENetworkParamSetUpdated(oldParams, newParams *networkv1alpha1.GKENetworkParamSet) error {
+	return c.updateGKENetworkParamSet(newParams)
+}
+
+func (c *Controller) updateGKENetworkParamSet(params *networkv1alpha1.GKENetworkParamSet) (rerr error) {
+	ctx := context.Background()
+	cidrs := []string{}
+
+	defer func() {
+		err := gkenetworkparamsetutil.UpdateGKENetworkParamSetStatus(ctx, c.networkClientset.NetworkingV1alpha1().GKENetworkParamSets(), params, cidrs)
+		if err != nil {
+			klog.Error(err, "failed to patch GKENetworkParamSetStatus")
+			rerr = multierr.Append(rerr, err)
+		}
+	}()
+
+	subnet, err := c.gceCloud.GetSubnetwork(c.gceCloud.Region(), params.Spec.VPCSubnet)
+	if err != nil {
+		klog.Errorf("unable to access subnetwork: %v", err)
+		return err
+	}
+
+	cidrs = extractRelevantCidrs(subnet, params)
+	return
+}
+
+func extractRelevantCidrs(subnet *compute.Subnetwork, params *networkv1alpha1.GKENetworkParamSet) []string {
```

Comment:
please add comment explaining the function, I'm curious about what makes a Cidr relevant ;)